client/src/modules/cohort/components/CohortHeader.jsx

```diff
@@ -0,0 +1,213 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import { CohortIcon } from 'assets/images/icons';
+import { updateCohort } from '../model/actions';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+
+class CohortHeader extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    const { details } = this.props;
+    const { name, description } = details;
+
+    this.state = {
+      description,
+      isDescriptionTextareaVisible: false,
+      isNameInputVisible: false,
+      name
+    };
+
+    this.descriptionTextarea = React.createRef();
+    this.nameInput = React.createRef();
+  }
+
+  componentDidMount() {
+    document.addEventListener('keypress', this.handleEnterPress);
+    document.addEventListener('mousedown', this.handleClick);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keypress', this.handleEnterPress);
+    document.removeEventListener('mousedown', this.handleClick);
+  }
+
+  handleClick = event => {
+    const {
+      isDescriptionTextareaVisible,
+      isNameInputVisible,
+      name
+    } = this.state;
+    const {
+      match: {
+        params: { id }
+      }
+    } = this.props;
+
+    if (
+      isDescriptionTextareaVisible &&
+      !this.descriptionTextarea.current.contains(event.target)
+    ) {
+      this.setState({ isDescriptionTextareaVisible: false });
+      this.handleDescriptionUpdate(id)();
+    }
+
+    if (
+      isNameInputVisible &&
+      name.length >= 1 &&
+      !this.nameInput.current.contains(event.target)
+    ) {
+      this.setState({ isNameInputVisible: false });
+      this.handleNameUpdate(id)();
+    }
+  };
+
+  handleEnterPress = event => {
+    const { isDescriptionTextareaVisible } = this.state;
+    const {
+      match: {
+        params: { id }
+      }
+    } = this.props;
+    const { code } = event;
+
+    if (code === 'Enter') {
+      const action = isDescriptionTextareaVisible
+        ? this.handleDescriptionUpdate
+        : this.handleNameUpdate;
+
+      action(id)();
+    }
+  };
+
+  handleNameInputVisibility = () => {
+    const { isNameInputVisible } = this.state;
+
+    this.setState({ isNameInputVisible: !isNameInputVisible });
+  };
+
+  handleDescriptionTextareaVisibility = () => {
+    const { isDescriptionTextareaVisible } = this.state;
+
+    this.setState({
```

Comment:
same as above:
this.setState(({ isDescriptionTextareaVisible }) => ({ isDescriptionTextareaVisible: ! isDescriptionTextareaVisible });